src/main/java/org/isf/menu/rest/UserSettingController.java

```diff
@@ -75,7 +80,15 @@ public UserSettingController(UserSettingMapper userSettingMapper, UserSettingMan
 	public ResponseEntity<UserSettingDTO> newUserSetting(@Valid @RequestBody UserSettingDTO userSettingDTO)
 					throws OHServiceException {
 		LOGGER.info("Attempting to create or update a UserSetting");
-		String userName = SecurityContextHolder.getContext().getAuthentication().getName();
+		String userName = userSettingDTO.getUser();
+		if (userName == null || userName.equals("")) {
+			userName = SecurityContextHolder.getContext().getAuthentication().getName();
+		} else {
+			if (userManager.getUserByName(userName) == null) {
+				throw new OHAPIException(new OHExceptionMessage("User not found."));
+			}
+		}
+		LOGGER.info("userName {}",userName);
```

Comment:
I don't get this change: BEFORE the setting was created always for the current user despite the passed `userName`; NOW if the `userName` is passed, it only checks if the users exists, which is good for admin, otherwise it just goes ahead with the creation/update, which is not good for other users, that can update any other user setting.

The issue states:

Aim of this issue is to fix the UserSettingController so it:

- allows to create and update the **current user’s own settings** `<-- not OK`
- allows admin create and update any user' settings `<-- OK`

So I think it is missing the check if the passed `userName` equals the current user.

I don't know how we could achieve **current user's own settings** while keeping **admin create and update any user' settings**, but I think we should enforce the first condition and maybe leave the second one as for now...

Make sense something like this?

```suggestion
		} else {
			if (!userName.equals(SecurityContextHolder.getContext().getAuthentication().getName()) {
			        throw new OHAPIException(new OHExceptionMessage("Not allowed."));
			}
		}
```